corehq/apps/export/models/deid_export.py

```diff
@@ -0,0 +1,42 @@
+import hashlib
+import uuid
+
+from django.db import models
+
+from corehq.const import ONE_DAY
+from corehq.util.quickcache import quickcache
+
+
+class DeIdMapping(models.Model):
+    domain = models.TextField(max_length=255)
+    hashed_value = models.TextField(max_length=32)
+    deid = models.UUIDField(default=uuid.uuid4)
+
+    class Meta:
+        indexes = [
+            models.Index(fields=['domain', 'hashed_value']),
+        ]
+
+    @classmethod
+    def get_deid(cls, value, doc, domain=None):
+        if doc is not None:
+            # use domain from the couch doc if one was passed in
+            domain = doc['domain']
+
+        return cls._get_deid(value, domain)
+
+    @classmethod
+    @quickcache(['value', 'domain'], timeout=90 * ONE_DAY)
+    def _get_deid(cls, value, domain):
+        hashed_value = cls._hash_value(value)
+        deid_mapping, __ = cls.objects.get_or_create(domain=domain, hashed_value=hashed_value)
+        return deid_mapping.deid
+
+    @staticmethod
+    @quickcache(['value'], timeout=90 * ONE_DAY)
```

Comment:
Why does the result of this function need to be cached?

My thoughts: The cache adds a network hit (to redis), but the function itself does not hit the network, so the cache would most likely make the function slower. Also, MD5 hashing should be very fast unless `value` is huge, but even in that case saving a huge value to the cache could be problematic.

Reply:
I cached it here without thinking about the additional network hit, just that the return value will stay the same for a given input so we "might as well" cache it. I assume you're probably correct that simply hashing the value and returning it would be more performant than caching it here, so I'll make that change.